app/models/solution/mentor_discussion_post.rb

```diff
@@ -13,4 +13,12 @@ class Solution::MentorDiscussionPost < ApplicationRecord
   validates :content_markdown, presence: true
 
   has_markdown_field :content
+
+  before_create do
+    self.uuid = SecureRandom.compact_uuid
+  end
+
+  def by_student?
+    discussion.student == author
```

Comment:
Use ids so we don't have to load the objects.

```suggestion
    discussion.student_id == author_id
```

Reply:
`student` isn't an ID here. It's part of a `has_one through` association.

Reply:
Then `discussion.student_id == author_id`?

Reply:
Discussion doesn't have a student ID since it's through the solution. It's going to look something like this:

```
  def by_student?
    discussion.solution.user_id == user_id
  end
```

We have no choice but to load the solution.

Reply:
Probably fine. If it becomes an issue you can "cache" or also add the user_id/student_id to the discussion.